util/topgen/merge.py

```diff
@@ -477,6 +512,61 @@ def no_device_in_range(xbar, f, t):
     return True
 
 
+def get_next_base_addr(addr, xbar, name):
+    """Return the least value of base_addr of the IP greater than addr
+
+    """
+    if isinstance(addr, str):
+        value = int(addr, 0)
+    else:
+        assert isinstance(addr, int)
+        value = addr
+
+    device_list = [
+        x for x in xbar["nodes"]
+        if x["type"] == "device" and not x["name"] == name
+    ]
+
+    try:
+        addrs = [a for r in device_list for a in r["addr_range"]]
+    except KeyError:
+        log.error("Address range is wrong.\n {}".format(
+            [x for x in device_list if not "addr_range" in x]))
+        raise SystemExit()
+
+    sorted_list = sorted(addrs, key=lambda k: int(k["base_addr"], 0))
+
+    gte_list = [x for x in sorted_list if int(x["base_addr"], 0) > value]
+
+    if len(gte_list) == 0:
+        return -1
+
+    return gte_list[0]
+
+
+def find_pow2_size(addr, next_value):
```

Comment:
Related to the previous comments by @tjaychen: Does it always make sense to find the greatest power of 2 value here? I.e. if `next_value` is set to `0x1_0000_0000`, this would find a very large power of two... How about we just look for the next power of two, i.e. something like:
```
aligned_size = 2**ceil(log2(region_size))
if aligned_size + base_addr > next_value:
  aligned_size = region_size
```

Reply:
yeah i tend to feel the closest power of 2 is better also... I think maybe it doesn't make a huge difference, since you already check for non-overlap with the closest device address..so i can't give you a very direct reason why I prefer the smaller instead of larger alignment. 

Let's use my made up example again, say we have two regions
0x4000_0000, 0x1000
0x4008_0000, 0x3000

Doesn't the existing code simplify this into one large region from 
0x4000_0000 to large number aligned to power of 2? 

Whereas it feels like 0x4000_0000 to 0x4010_0000 would be sufficient.

Reply:
The reason the code tries to find the maximum value is, it can be reduced to use smaller bits of address. with bigger size (if it is power of 2), the logic can compare only upper bits of the address, the current RTL isn't coded that way though...

Reply:
o okay that's fair, so you're just reducing the number of bits on the top most side.  I'm not sure how much it matters at the end though, because this really only applies if you have a xbar attached at the end of the current xbar....but shrinking smaller doesn't confer any special advantage either.

Can we think of any security reason why we wouldn't want to? The address range "enlarging" really applies to xbar connections right? So on the secondary xbar that connects to devices, it would still be very precise, is that right?

Reply:
Yes. The precise check is done at the lower boundary.